### ISSUE TITLE
Api backend example excessively permissive

### ISSUE THREAD

**Issue by dave-graham** [awslabs/serverless-application-model#757]
Hi,

Yesterday I found https://github.com/awslabs/serverless-application-model/blob/master/docs/policy_templates.rst for the first time, despite using SAM on a day to day basis for around 6 months. It's very powerful, but I clearly missed the link to it on here https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md

I think that one of the reasons that I may have missed it is I used the example at  https://github.com/awslabs/serverless-application-model/blob/master/examples/2016-10-31/api_backend/template.yaml as the basis for a number of rest apis we have built out.

Should the following lines:

```
Policies: AmazonDynamoDBReadOnlyAccess

Policies: AmazonDynamoDBFullAccess

Policies: AmazonDynamoDBFullAccess
```

Be changed to:

```
- DynamoDBReadPolicy:
	TableName: !Ref Table

- DynamoDBCrudPolicy:
	TableName: !Ref Table

- DynamoDBCrudPolicy:
	TableName: !Ref Table
```

This will ensure that for a simple api, developers are following a good pattern for dynamoDB service access.

**Comment by keetonian** [comment]
Yes, that is correct. This is also good feedback about Policy Templates- they are a great feature of SAM, but if it took you this long to discover them, it means we need to somehow surface them better in the documentation.

**Comment by jlhood** [comment]
@dave-graham Agreed we should definitely change that example. Would you mind submitting the PR?

**Comment by dave-graham** [comment]
@jlhood have done 

@keetonian throwing an idea out there, is it worth having a best practices page in the documentation that links to or covers things like:

[Policy templates](https://github.com/awslabs/serverless-application-model/blob/master/docs/policy_templates.rst)
[Safe deployments](https://github.com/awslabs/serverless-application-model/blob/master/docs/safe_lambda_deployments.rst)
[Approaches to environment separation](https://github.com/awslabs/serverless-application-model/issues/220#issuecomment-349054599) - I know this overlaps with sam cli, but we've needed to pass environment parameters into the sam template to achieve this, as we only have access to 1 aws instance
[Backups + recovery](https://github.com/awslabs/serverless-application-model/issues/714) - once I get round to implementing
[Dynamo encryption](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-dynamodb-table-ssespecification.html)
[Scaling Dynamo](https://github.com/awslabs/serverless-application-model/pull/705) - or once it goes live

**Comment by keetonian** [comment]
@dave-graham I personally think so; there are parts of the docs that have useful information that are hard to find. Having a best practices page that highlights development practices and useful features would be a great idea. Go ahead and start an issue so we can discuss and gather the areas we think that should be documented in a best practices section of the docs.
